mmv1/third_party/terraform/resources/resource_compute_disk_async_replication.go.erb

```diff
@@ -0,0 +1,317 @@
+<% autogen_exception -%>
+package google
+<% if version != "ga" -%>
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+	"log"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+<% if version == "ga" -%>
+	"google.golang.org/api/compute/v1"
+<% else -%>
+	compute "google.golang.org/api/compute/v0.beta"
+<% end -%>
+)
+
+func ResourceComputeDiskAsyncReplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDiskAsyncReplicationCreate,
+		Read:   resourceDiskAsyncReplicationRead,
+		Update: resourceDiskAsyncReplicationUpdate,
+		Delete: resourceDiskAsyncReplicationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(300 * time.Second),
+			Update: schema.DefaultTimeout(300 * time.Second),
+			Delete: schema.DefaultTimeout(300 * time.Second),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"primary_disk": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				Description:      `Primary disk for asynchronous replication.`,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"secondary_disk": {
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Description: `Secondary disk for asynchronous replication.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"disk": {
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+							Description: `Secondary disk for asynchronous replication.`,
+							DiffSuppressFunc: compareSelfLinkOrResourceName,
+						},
+						"state": {
+							Type:         schema.TypeString,
+							Computed:     true,
+							Description:  `Output-only. Status of replication on the secondary disk.`,
+						},
+					},
+				},
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func getComputeClientForAsyncReplication(d *schema.ResourceData, meta interface{}) (*compute.Service, error) {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := generateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return nil, err
+	}
+
+	clientCompute := config.NewComputeClient(userAgent)
+	return clientCompute, nil
+}
+
+func getDiskFromConfig(disk string, d *schema.ResourceData, meta interface{}) (zonal bool, zv *ZonalFieldValue, rv *RegionalFieldValue, resourceId string, err error) {
```

Comment:
Should `zonal bool` be omitted? It seems checking rv == nil should give the same information

Reply:
Omitted.